pkg/rc/farm.go

```diff
@@ -277,12 +278,17 @@ func (rcf *Farm) failsafe(rcFields []rcstore.RCLockResult) {
 		}
 		panic("No RCs are scheduled at all. Create one RC to enable the farm. Panicking to escape a potentially bad situation.")
 	}
+
 	globalReplicaCount := 0
 	for _, rc := range rcFields {
-		if rc.ReplicasDesired > 0 {
+		if rc.ReplicasDesired == nil {
+			panic(fmt.Sprintf("RC %s is missing replicas_desired count", rc.ID))
+		}
+
+		if *rc.ReplicasDesired > 0 {
 			return
 		}
-		globalReplicaCount += rc.ReplicasDesired
+		globalReplicaCount += *rc.ReplicasDesired
 	}
 	if globalReplicaCount == 0 {
```

Comment:
I know you didn't write this, but it strikes me that at this point `globalReplicaCount` doesn't matter - the function would have returned if any was nonzero, so if we get to this point, `globalReplicaCount` is guaranteed to be zero and the counter and `if` might as well be removed.